crates/primitives/src/chain_spec.rs

```diff
@@ -211,23 +259,29 @@ impl From<EthersGenesis> for ChainSpec {
             (Hardfork::London, genesis.config.london_block),
             (Hardfork::ArrowGlacier, genesis.config.arrow_glacier_block),
             (Hardfork::GrayGlacier, genesis.config.gray_glacier_block),
-            (Hardfork::MergeNetsplit, genesis.config.merge_netsplit_block),
         ];
 
-        let configured_hardforks = hardfork_opts
+        let mut configured_hardforks = hardfork_opts
             .iter()
-            .filter_map(|(hardfork, opt)| opt.map(|block| (*hardfork, block)))
+            .filter_map(|(hardfork, opt)| opt.map(|block| (*hardfork, ForkKind::Block(block))))
             .collect::<BTreeMap<_, _>>();
 
+        if let Some(paris_ttd) = genesis.config.terminal_total_difficulty {
+            configured_hardforks.insert(
+                Hardfork::MergeNetsplit,
+                ForkKind::TTD(TerminalTotalDifficulty::new(
+                    paris_ttd.into(),
+                    genesis.config.merge_netsplit_block,
+                )),
+            );
```

Comment:
similar correction to the other comments about `Hardfork::MergeNetsplit` - this should probably be removed and replaced by re-introducing and setting `paris_ttd` in `ChainSpec`